mslice/plotting/globalfiguremanager.py

```diff
@@ -129,7 +138,7 @@ def _new_figure(cls, num=None):
             num = 1
             while any([num == existing_fig_num for existing_fig_num in cls._figures.keys()]):
                 num += 1
-        new_fig = PlotFigureManagerQT(num, GlobalFigureManager)
+        new_fig = QAppThreadCall(PlotFigureManagerQT)(num, GlobalFigureManager)
```

Comment:
So it was just a matter of wrapping in a Mantid `QAppThreadCall` in the end?

Reply:
This and a similar `QAppThreadCall(mainloop)()` in the mslice `Show` method yes.